docs-developer/docker.md

```diff
@@ -0,0 +1,92 @@
+# Docker
+
+In production the profiler server runs in a docker container. In this document
+we'll describe some of the basic commands around creating and handling this
+docker container.
+
+## Basic commands
+### Build the docker image
+
+The [Dockerfile](../Dockerfile) describes the image. It's not meant to be run
+directly because it assumes some build steps have been run before building the
+image.
+
+Therefore the correct way to build the image is to run the command:
+```
+yarn docker
+```
+
+The generated image has a tag from the name and the version in package.json,
+currently `profiler-server:1.0.0`.
+
+The command `yarn build:docker` only runs the image building step. This is
+normally not run directly either, but can be used when working on the
+Dockerfile.
+
+Using `yarn build:docker --pull` will update the base image, it's useful to do
+it from time to time.
+
+### Run the docker image in a container
+
+We can run the last built image in a new container with the command:
+```
+yarn docker:run
+```
+The app should listen on port 8000.
+
+Pressing Ctrl-C will gracefully close the app and the container, then delete the
+container.
+
+Pressing Ctrl-P CTRL-Q will put the container in background. You can use `docker
+container ls` and `docker container stop <name>` to stop it then.`
+
+## More information and troubleshooting
+### Running the docker image
+`yarn docker:run` is a shortcut for:
+```
+docker run -t -i -p8000:8000 --rm profiler-server:1.0.0
```

Comment:
ah, I'm used to stick the value to its parameters because sometimes (rarely) some commands only understand this :)
I can definitely change it though!